packages/react/src/ActionList/Item.tsx

```diff
@@ -281,20 +306,44 @@ export const Item = React.forwardRef<HTMLLIElement, ActionListItemProps>(
       id: itemId,
     }
 
-    const containerProps = _PrivateItemWrapper ? {role: itemRole ? 'none' : undefined} : menuItemProps
+    let containerProps
+    let wrapperProps
 
-    const wrapperProps = _PrivateItemWrapper ? menuItemProps : {}
+    if (buttonSemantics) {
+      containerProps = _PrivateItemWrapper
+        ? {role: itemRole ? 'none' : undefined, ...props}
+        : // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+          (validRole && {...menuItemProps, ...props, ref: forwardedRef}) || {}
+
+      wrapperProps = _PrivateItemWrapper
+        ? menuItemProps
+        : !validRole && {
+            ...menuItemProps,
+            ...props,
+            styles: merge<BetterSystemStyleObject>(styles, sxProp),
+            ref: forwardedRef,
+          }
+    } else {
+      containerProps = _PrivateItemWrapper ? {role: itemRole ? 'none' : undefined} : {...menuItemProps, ...props}
+      wrapperProps = _PrivateItemWrapper ? menuItemProps : {}
+    }
 
     return (
       <ItemContext.Provider
         value={{variant, disabled, inactive: Boolean(inactiveText), inlineDescriptionId, blockDescriptionId}}
       >
         <LiBox
-          ref={forwardedRef}
-          sx={merge<BetterSystemStyleObject>(styles, sxProp)}
+          ref={buttonSemantics || validRole ? forwardedRef : null}
```

Comment:
It switches between the inner `<button>` and the list item depending on the context. When the `<button>` is applied, we pass the props and `forwardedRef` to it, whereas if it's a list item only (i.e. `validRole` is true), then we provide the `forwardedRef` to that list item. I did it this way so that the `forwardedRef` will always be on the thing that you can pass props to.